packages/application/src/connectionlost.ts

```diff
@@ -1,11 +1,14 @@
 // Copyright (c) Jupyter Development Team.
 // Distributed under the terms of the Modified BSD License.
 
-import { showErrorMessage } from '@jupyterlab/apputils';
+import { Dialog, showDialog } from '@jupyterlab/apputils';
 import { ServerConnection, ServiceManager } from '@jupyterlab/services';
 import { ITranslator, nullTranslator } from '@jupyterlab/translation';
 import { IConnectionLost } from './tokens';
 
+let displayConnectionLost = true;
+let serverConnectionLost: Promise<void | Dialog.IResult<string>> | undefined;
```

Comment:
This is just stylistic, but I would suggest to wrap this in a `Private` namespace and move to the end of the file. See how this is done on in the clipboard package for example:

https://github.com/jupyterlab/jupyterlab/blob/ae123cb82015540804964f76ba4e56f21799ea27/packages/apputils/src/clipboard.ts#L99-L107

This will makes it clear these variables are private, and also allow you to write a structured docstring for each of them :)